FreeRTOS/Test/Target/tests/smp/schedule_highest_priority/schedule_highest_priority.c

```diff
@@ -0,0 +1,193 @@
+
+#if ( configMAX_PRIORITIES <= configNUMBER_OF_CORES )
+    #error This test creates tasks with different priority, requires configMAX_PRIORITIES to be larger than configNUMBER_OF_CORES.
+#endif /* if configNUMBER_OF_CORES != 2 */
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test case "Only One Task Enter Critical".
+ */
+void Test_ScheduleHighestPirority( void );
+
+/**
+ * @brief Function that implements a never blocking FreeRTOS task.
+ */
+static void prvEverRunningTask( void * pvParameters );
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Handle of the testRunner task.
+ */
+static TaskHandle_t xTestRunnerTaskHanlde;
+
+/**
+ * @brief Handles of the tasks created in this test.
+ */
+static TaskHandle_t xTaskHanldes[ configNUMBER_OF_CORES ];
+/*-----------------------------------------------------------*/
+
+static void prvEverRunningTask( void * pvParameters )
+{
+    int i = 0;
+    int currentTaskIdx = ( int ) pvParameters;
+    eTaskState xTaskState;
+
+    for( i = 0; i < currentTaskIdx; i++ )
+    {
+        xTaskState = eTaskGetState( xTaskHanldes[ i ] );
+
+        /* Tasks created in this test are of descending priority order. For example,
+         * priority of T0 is higher than priority of T1. A lower priority task is able
+         * to run only when the higher priority tasks are running. Verify that higher
+         * priority tasks is of running state. */
+        if( eRunning != xTaskState )
+        {
+            while( xTestRunnerTaskHanlde == NULL )
+            {
+                /* Waiting testRunner to update xTestRunnerTaskHanlde. */
+                vTaskDelay( pdMS_TO_TICKS( 10 ) );
+            }
+
+            /* Generate error code corresponds to task index.
+             * Task 0: 0x10
+             * Task 1: 0x11
+             * ... */
+            ( void ) xTaskNotify( xTestRunnerTaskHanlde, ( uint32_t ) ( currentTaskIdx + 0x10 ), eSetValueWithoutOverwrite );
+        }
+    }
+
+    /* If the task is the last task, then we finish the check because all tasks are checked. */
+    if( currentTaskIdx == ( configNUMBER_OF_CORES - 1 ) )
+    {
+        ( void ) xTaskNotify( xTestRunnerTaskHanlde, ( uint32_t ) ( pdPASS ), eSetValueWithoutOverwrite );
+    }
+
+    for( ; ; )
+    {
+        /* Always running, put asm here to avoid optimization by compiler. */
+        __asm volatile ( "nop" );
+    }
+}
+/*-----------------------------------------------------------*/
+
+void Test_ScheduleHighestPirority( void )
+{
+    uint32_t ulNotifiedValue;
+    TickType_t xStartTick = xTaskGetTickCount();
+
+    xTestRunnerTaskHanlde = xTaskGetCurrentTaskHandle();
+
+    if( xTaskNotifyWait( 0x00, ULONG_MAX, &ulNotifiedValue, pdMS_TO_TICKS( TEST_TIMEOUT_MS ) ) != pdTRUE )
+    {
+        /* Timeout, test fail. */
+        TEST_ASSERT_TRUE( pdFALSE );
+    }
+    else
+    {
+        TEST_ASSERT_EQUAL_INT( pdPASS, ulNotifiedValue );
+    }
```

Comment:
```suggestion
    xReturn = xTaskNotifyWait( 0x00, ULONG_MAX, &ulNotifiedValue, pdMS_TO_TICKS( TEST_TIMEOUT_MS );

    TEST_ASSERT_EQUAL( pdTRUE, xReturn );
    TEST_ASSERT_EQUAL_INT( pdPASS, ulNotifiedValue );
```
Please also declare xReturn.